### PR TITLE
WM-255: coverage, hooks, prettier/lint

### PR DESCRIPTION
**Pull request by alxyuu**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,21 +35,48 @@
...
   },
   "jest": {
+    "coverageReporters": [
+      "json",
+      "lcov",
+      "text",
+      "clover"
+    ],
+    "coverageThreshold": {
+      "global": {
```
Comment (macleanevans) [inline package.json]: We are going to drop this on the main repo. Do you want to keep it 100 in here?

Comment (alxyuu) [reply package.json]: is reagan's testing proposal already published? I mainly wanted it for my own benefit so I wouldn't forget to test something, given these are shared utils my opinion would be that they should have complete coverage.

that being said I'm not opposed to adopting whatever the new standard on web will be

File: src/FadoClient.js [hunk 2]
```diff
@@ -18,6 +20,7 @@ export default class FadoClient {
       this.ws.onmessage = this.forwardMessage.bind(this);
       this.ws.onclose = this.reconnect.bind(this);
 
+      // eslint-disable-next-line no-underscore-dangle
```
Comment (alxyuu) [inline src/FadoClient.js]: is there a better way to do this? __DEV__ is a flag on react native

File: package.json (hunk 1 above)
Comment (macleanevans) [reply package.json]: https://rigup.atlassian.net/wiki/spaces/PROD/pages/154271991/Make+React+Code+Coverage+Actionable+-+11+14+19

We haven't made a final decision yet (probably tomorrow) but all the responses have been positive.

Comment (alxyuu) [reply package.json]: the way I see it, the nature of the code being tested is very different between util and web (shared libraries other feature depend on versus more ui feature work), so I would say the need for stability in shared libraries needs to be greater. it also helps that the cadence of changes to util should be lower. 

the I way I see the proposal, it's more a removal of requirements that aren't significantly benefiting anyone in an effort to unblock us and help us move forward while taking our time to develop better testing standards. maybe I'm misunderstanding the proposal though

@reaganthomas any thoughts?

**Pull request merged**